docs/snapshot-testing.md

```diff
@@ -0,0 +1,113 @@
+# Snapshot Testing
+
+## What is Snapshot Testing?
+
+Snapshot testing is an extension of usual expected-vs-actual assertion where the following conditions apply:
+
+1. The expected object ("snapshot" afterwards) is too large to be contained in a few lines of code (e.g. a web page HTML content)
+1. The snapshot should be tolerant to some degree of data variation (e.g. date and time of test execution)
+1. (Optional) There is an automated way to update the snapshot based on the state of the actual object at any point of time
+
+In the case of TEAMMATES, snapshot testing is useful for the following comparisons:
+
+1. Actual rendered web page vs expected web page content/DOM structure
+1. Actual email generated by the system vs expected email content
+1. Actual CSV generated by the system vs expected CSV content
+
+## How does Snapshot Testing work?
+
+When testing (i.e. doing the assertions), snapshot testing is the same as normal assertion.
+
+The flagship feature of snapshot testing is the **auto-update mode**. Essentially, this _reverses_ the process of testing, i.e. using the _actual_ generated object to overwrite the snapshot provided in the test. To remove redundancy, even if auto-update mode is enabled, this overwriting procedure only happens when a test fails during the test run.
+
+For such testing method to be efficient, before the changes are committed, a *manual* (by the developer) verification to ensure only the intended changes have occurred is mandatory.
+
+## How do we use Snapshot Testing?
+
+For the web page comparison, the tests are done in the front-end using Jest. [Jest has native support for snapshot testing (in fact, that is where the name is obtained from!)](https://jestjs.io/docs/en/snapshot-testing). Auto-update mode is activated by pressing `u` when running Jest under watch mode.
```

Comment:
Feel like it'll be easier if we also link to our own existing snapshot tests, so that contributors can where to write the tests, and see how the data is passed into the angular component for rendering the page.